utils.c

```diff
@@ -18,10 +18,21 @@
 #include <dirent.h>
 #include <errno.h>
 #include <err.h>
+#include <unistd.h>
 
 #include "version.h"
 #include "utils.h"
 
+void restrict_fsaccess(const char **rootdir)
+{
+	if (!chroot(*rootdir)) {
+		*rootdir = "/";
```

Comment:
Let's make `*rootdir = "";`, move the new comment in `full_fn_from_number()` to `restrict_fsaccess()`; adjust the comment as needed. Then drop the patch in `full_fn_from_number().

Reply:
Err... are you still _open to code that makes chroot() available on all platforms but Cygwin_? 😅

Reply:
Continue practicing your funny side, but don't quit your day job anytime soon. :)

Reply:
IMAO `*rootdir` shall reflect the _actual_ root dir. `full_fn_from_number()` may be not the only consumer and shall just render clean paths IMAO (/, not //).

Reply:
OK, forget chroot(2) for a second. If you actually target / on Cygwin, do you expect F3 to write to / or // (different dirs)?

Reply:
`full_fn_from_number()` assumes that it is always needed to add '/' to extend a path. I maintain/enforce this assumption in the software I write to avoid checking the end of the path. So making "*rootdir` the empty string fits well into F3.

Reply:
## An offer you can’t refuse

1. F3 unconditionally chdir(2)s the target directory
2. F3 uses relative paths (1.h2w)

So this complete problem would just go away and btw. chdir(2) would "occupy" the device for F3's lifetime. So no umount and no other bad surprises.

What does this sound like?

Reply:
Your offer is good; I like it is clean. But I'm not confident that I won't have to patch the code to accommodate some problem for another user. So until I have the possibility of reverting the code to the old way for some platforms eliminated, I prefer the way I've suggested in the review.

My perspective is different because once you're done, you'll go away. Whereas I'll be around replying to other users and maintaining the code. To make my point more concrete, once we merge and the code has not reverted to the old way for a "long" time (ideally 1 year), I'll accept a pull request to work out your irrefutable offer.

Reply:
It looks good, please go ahead and test it.